client/src/components/TreeImageScrubber.js

```diff
@@ -93,63 +105,96 @@ function TreeImageScrubber({ classes, ...props }) {
       });
   };
 
+  const setIsLoading = loading => {
+    state.isLoading = loading;
+  };
+
+  const loadMoreTreeImages = () => {
+    setIsLoading(true);
+    const nextPage = state.pagesLoaded + 1;
+    const pageParams = {
+      page: nextPage,
+      rowsPerPage: state.pageSize
+    };
+    getTreeImages(pageParams)
+      .then(result => {
+        setIsLoading(false);
+        state.pagesLoaded = nextPage;
+        dispatch({ type: "loadMoreTreeImages", treeImages: result });
+      })
+      .catch(error => {
+        // no more to load!
+        state.noMoreTreeImagesToLoad = true;
+      });
+  };
+
+  const handleScroll = e => {
+    if (
+      state.isLoading ||
+      (scrollContainerRef &&
+        Math.floor(scrollContainerRef.scrollTop) !==
```

Comment:
floor because these javascript height properties might return decimals..
the behavior is fairly not buggy.. ;)